File: hyeyoung/week12/로또의최고순위와최저순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int win = 0; // 맞춘 번호
        int zero = 0; // 알아볼 수 없는 번호
        
        for(int i : lottos){
            // 번호를 알아볼 수 없는 경우
            if(i == 0) zero++;
            else {
                for(int j : win_nums){  
                    if(i == j){
                        win++;
                        break;
                    }
                }
            }
        }
        
        answer[0] = Math.min(7 - (win + zero), 6);
        answer[1] = Math.min(7 - win, 6);
        return answer;
    
    }
}

File: SiHyeon/week12/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums)
{
    vector<int> answer = {0, 0};
    // 0이 아닌 값과 로또 점수를 비교
    // 모든 0이 맞으면 최고 순위
    // 모든 0이 틀리면 최저 순위
    int match = 0; // 맞는 갯수
    for (auto num : lottos)
    {
        if (find(win_nums.begin(), win_nums.end(), num) != win_nums.end())
        {
            match++;
        }
    }
    answer[1] = (6 - match) >= 5 ? 6 : 6 - match + 1; // 1,0개 맞으면 6등처리
    match += count(lottos.begin(), lottos.end(), 0);  // 0이 다 맞는 경우
    answer[0] = (6 - match) >= 5 ? 6 : 6 - match + 1;
    return answer;
}